unstable/wlr-export-dmabuf-unstable-v1.xml

```diff
@@ -0,0 +1,252 @@
+    DEALINGS IN THE SOFTWARE.
+  </copyright>
+
+  <description summary="a protocol for low overhead screen content capturing">
+    An interface to capture surfaces in an efficient way.
+    Overall usage:
+
+    1.) client registers with zwlr_screencontent_manager_v1
+    2.) server sends client info about surfaces via "receive_surface_info"
+    3.) client subscribes to capture a surface via the "capture" requests
+    4.) server sends client events via the "zwlr_screencontent_frame" interface
+    5.) client finishes and informs server via the "frame_destroy" event
+    6.) client optionally resubscribes via repeating steps 3.) through 5.)
+  </description>
+
+  <interface name="zwlr_export_dmabuf_frame_v1" version="1">
+    <description summary="a frame ready for readout">
+      This object represents a frame which is ready to have its resources
+      fetched and used.
+
+      The receive callback shall be called first, followed by the "object"
+      callback once per dmabuf object or the "layer" callback, once per dmabuf
+      layer. The "plane" callback shall only be called after the "layer"
+      callback corresponding to the layer the plane belongs to has been called
+      Finally, the "ready" event is called to indicate that all the data has
+      been made available for readout, as well as the time at which presentation
+      happened at. The ownership of the frame is passed to the client, who's
+      responsible for destroying it via the "destroy" event once finished.
+      The data the API describes has been based off of what
+      VASurfaceAttribExternalBuffers contains.
+      All frames are read-only and may not be written into or altered.
+    </description>
+
+    <enum name="flags">
+      <description summary="frame flags">
+        Special flags that must be respected by the client.
+        Transient frames indicate short lifetime frames (such as swapchain
+        images from external clients). Clients are advised to copy them and do
+        all processing outside of the "ready" event.
+      </description>
+      <entry name="transient" value="0x1" since="1"/>
+    </enum>
+
+    <event name="frame">
+      <description summary="main callback">
+        Main callback supplying the client with information about the frame,
+        as well as an object to serve as context for destruction. Always called
+        first before any other events.
+
+        The "transform" argument describes the orientation needed to be applied
+        to correctly orient the buffer. For example, a buffer rotated by 90
+        degrees will have a value of "3" here, corresponding to the need to
+        apply a 270 degree transpose to correctly present the buffer.
+      </description>
+      <arg name="width" type="uint"
+           summary="frame width, scaling factor included"/>
+      <arg name="height" type="uint"
+           summary="frame height, scaling factor included"/>
+      <arg name="scale" type="uint"
```

Comment:
How can we handle fractional scaling?

I'm not a fan of making this a `fixed`.

For outputs, the client can read all the necessary transform information from wl_output and xdg-output. The integral scale and the transform are in wl_output, the logical size of the output (which gives a x- and a y-axis fractional scale) is in xdg-output. So maybe the client can do a better job using that instead of our frame properties.

We still need to support clients. Clients can only apply integral scaling and transformations to their buffers, not fractional scaling. But some other protocols (e.g. wp_viewporter) might add new transformations to client buffers, not sure how to handle those.